 <launch>

 	<!-- SENSORS -->
  	<arg name="enableImu"   default="true"/>
	<arg name="enableDepth" default="true"/>
	<arg name="enableDvl"   default="true"/>
	<arg name="enableGps"   default="true"/>
	<arg name="enableModem" default="true"/>
	<arg name="enableVO"    default="false"/>
  	<arg name="enableSafety" default="false"/>

	<!-- BAGFILES SURVEY BIG-->
	<arg name="bagfileRoute"  default="/media/eric/40E5-6688/bagfiles/2016_06_13_sant_feliu/15/" />
	<arg name="bagfileName0"  default="$(arg bagfileRoute)2016-06-15-11-54-09_0.bag" />
	<arg name="bagfileName1"  default="$(arg bagfileRoute)2016-06-15-11-55-48_1.bag" />
	<arg name="bagfileName2"  default="$(arg bagfileRoute)2016-06-15-11-57-27_2.bag" />
	<arg name="bagfileName3"  default="$(arg bagfileRoute)2016-06-15-11-59-06_3.bag" />
	<arg name="bagfileName4"  default="$(arg bagfileRoute)2016-06-15-12-00-45_4.bag" />
	<arg name="bagfileName5"  default="$(arg bagfileRoute)2016-06-15-12-02-24_5.bag" />
	<arg name="bagfileName6"  default="$(arg bagfileRoute)2016-06-15-12-04-03_6.bag" />
	<arg name="bagfileName7"  default="$(arg bagfileRoute)2016-06-15-12-05-42_7.bag" />
	<arg name="bagfileName8"  default="$(arg bagfileRoute)2016-06-15-12-07-21_8.bag" />
	<arg name="bagfileName9"  default="$(arg bagfileRoute)2016-06-15-12-09-00_9.bag" />
	<arg name="bagfileName10" default="$(arg bagfileRoute)2016-06-15-12-10-39_10.bag" />
	<arg name="bagfileName11" default="$(arg bagfileRoute)2016-06-15-12-12-18_11.bag" />
	<arg name="bagfileName12" default="$(arg bagfileRoute)2016-06-15-12-13-57_12.bag" />
	<arg name="bagfileName13" default="$(arg bagfileRoute)2016-06-15-12-15-36_13.bag" />
	<arg name="bagfileNames" default="$(arg bagfileName0) $(arg bagfileName1) $(arg bagfileName2) $(arg bagfileName3) $(arg bagfileName4) $(arg bagfileName5) $(arg bagfileName6) $(arg bagfileName7) $(arg bagfileName8) $(arg bagfileName9) $(arg bagfileName10) $(arg bagfileName11) $(arg bagfileName12) $(arg bagfileName13)" />


	<!-- BAGFILES SURVEY 7x7-->
<!--  	<arg name="bagfileRoute"  default="/media/eric/40E5-6688/bagfiles/2016_06_13_sant_feliu/14/" />
	<arg name="bagfileName0"  default="$(arg bagfileRoute)2016-06-14-13-50-03_0.bag" />
	<arg name="bagfileName1"  default="$(arg bagfileRoute)2016-06-14-13-51-42_1.bag" />
	<arg name="bagfileName2"  default="$(arg bagfileRoute)2016-06-14-13-53-21_2.bag" />
	<arg name="bagfileName3"  default="$(arg bagfileRoute)2016-06-14-13-55-00_3.bag" />
	<arg name="bagfileName4"  default="$(arg bagfileRoute)2016-06-14-13-56-39_4.bag" />
	<arg name="bagfileName5"  default="$(arg bagfileRoute)2016-06-14-13-58-18_5.bag" />
	<arg name="bagfileName6"  default="$(arg bagfileRoute)2016-06-14-14-01-17_6.bag" />
	<arg name="bagfileName7"  default="$(arg bagfileRoute)2016-06-14-14-04-31_7.bag" />
	<arg name="bagfileNames" default="$(arg bagfileName0) $(arg bagfileName1) $(arg bagfileName2) $(arg bagfileName3) $(arg bagfileName4) $(arg bagfileName5) $(arg bagfileName6) $(arg bagfileName7)" />
 -->
	<!-- BAGFILES SURVEY ENSAIMADA-->
<!-- 	<arg name="bagfileRoute"  default="/media/eric/40E5-6688/bagfiles/2016_06_13_sant_feliu/15/" />
	<arg name="bagfileName0"  default="$(arg bagfileRoute)2016-06-14-13-09-37_0.bag" />
	<arg name="bagfileName1"  default="$(arg bagfileRoute)2016-06-14-13-11-16_1.bag" />
	<arg name="bagfileName2"  default="$(arg bagfileRoute)2016-06-14-13-12-55_2.bag" />
	<arg name="bagfileName3"  default="$(arg bagfileRoute)2016-06-14-13-14-34_3.bag" />
	<arg name="bagfileName4"  default="$(arg bagfileRoute)2016-06-14-13-16-13_4.bag" />
	<arg name="bagfileName5"  default="$(arg bagfileRoute)2016-06-14-13-17-52_5.bag" />
	<arg name="bagfileName6"  default="$(arg bagfileRoute)2016-06-14-13-19-31_6.bag" />
	<arg name="bagfileName7"  default="$(arg bagfileRoute)2016-06-14-13-21-10_7.bag" />
	<arg name="bagfileName8"  default="$(arg bagfileRoute)2016-06-14-13-22-49_8.bag" />
	<arg name="bagfileName9"  default="$(arg bagfileRoute)2016-06-14-13-24-28_9.bag" />
	<arg name="bagfileName10" default="$(arg bagfileRoute)2016-06-14-13-26-07_10.bag" />
	<arg name="bagfileNames" default="$(arg bagfileName0) $(arg bagfileName1) $(arg bagfileName2) $(arg bagfileName3) $(arg bagfileName4) $(arg bagfileName5) $(arg bagfileName6) $(arg bagfileName7) $(arg bagfileName8) $(arg bagfileName9) $(arg bagfileName10)" /> -->

	<!-- LOAD BAGFILES-->
	<arg name="topics" default="/sensors/dvl_raw /sensors/vo_raw /sensors/imu_raw /sensors/depth_raw /sensors/gps_raw /sensors/modem_delayed_acoustic /control/trajectory_path /control/waypoint_marker /ekf_odom/odometry"/>

	<param name="use_sim_time" value="true"/>
	<node name="rosbag" pkg="rosbag" type="play" args=" --clock $(arg bagfileNames) --topics $(arg topics)" output="screen">
		<remap from="/ekf_odom/odometry" to="/modem/odometry"/>
	</node> 


 	<!-- NED origin -->
  	<arg name="place" default="sant_feliu.yaml"/>
  	<arg name="placesDir" default="$(find turbot)/config/places"/>
  	<rosparam command="load" file="$(arg placesDir)/$(arg place)"/>

	<!-- TFs -->
	<rosparam command="load" file="$(find turbot)/config/frames.yaml"/>

	<include file="$(find turbot)/launch/modules/transforms.launch">
	  <arg name="suffix" value="robot"/>
	</include>

	<node  name="pose_tf_broadcaster" pkg="sensors" type="pose_tf_broadcaster" respawn="true" output="screen" >
	  <remap from="/imu" to="/sensors/imu"/>
	</node>

	<!-- USBL POSITION -->
<!-- 	<node pkg="usbl_position" name="usbl_position" type="usbl_position" output="screen">
	  <remap from="/usbl_position/modem_delayed" to="/sensors/modem_delayed_acoustic"/>
	  <remap from="/usbl_position/buoy_ned" to="/sensors/buoy_ned"/>
	</node> -->

	<!-- USBL PROJECTION -->
 	<node pkg="usbl_position" name="usbl_projection" type="usbl_projection" output="screen">
 		<remap from="/ekf_odom/odometry" to="/modem/odometry"/>
 		<param name="usbl/sync_time_th"   value="0.1" />
 		<param name="usbl/sync_prop_th"   value="1.2" />
 		<param name="usbl/sync_disp_th"   value="0.8" />
 		<param name="usbl/cov"   		  value="0.5" />
	</node>

	<!-- USBL PLOT -->
	<node pkg="usbl_position" name="plot_path" type="plot_path">
	</node>

	<!-- Sensor aggregator -->
	<rosparam command="load" file="$(find turbot)/config/sensor_aggregator.yaml"/>
	<node  name="sensors" pkg="sensors" type="sensor_aggregator" output="screen" respawn="true">
	  <param name="enable_safety" value="$(arg enableSafety)" />
	  <param name="depth/enable"  value="$(arg enableDepth)" />
	  <param name="imu/enable"    value="$(arg enableImu)" />
	  <param name="dvl/enable"    value="$(arg enableDvl)" />
	  <param name="gps/enable"    value="$(arg enableGps)" />
	  <param name="modem/enable"  value="$(arg enableModem)" />
	  <param name="vo/enable"     value="$(arg enableVO)" />
	</node>

	<!-- Nav status -->
	<node pkg="sensors" type="nav_status" name="nav_status" respawn="true" output="screen">
	  <remap from="odometry" to="/ekf_map/odometry"/>
	  <remap from="altitude" to="/altitude_estimator/altitude"/>
	  <remap from="imu" to="/sensors/imu"/>
	  <param name="places_dir" value="$(arg placesDir)" />
	  <remap from="nav_status/nav_sts" to="/navigation/nav_sts"/>
	</node>


	<!-- Robot localization -->
	<rosparam command="load" file="$(find turbot)/config/robot_localization.yaml"/>
	<node pkg="robot_localization" type="ekf_localization_node" name="ekf_odom" respawn="true" output="screen" clear_params="true">
	  <remap from="odometry/filtered" to="/ekf_odom/odometry"/>
	  <remap from="/set_pose" to="/ekf_odom/set_pose"/>
	</node>
	<node pkg="robot_localization" type="ekf_localization_node" name="ekf_map" respawn="true" output="screen" clear_params="true">
	  <remap from="odometry/filtered" to="/ekf_map/odometry"/>
	  <remap from="/set_pose" to="/ekf_map/set_pose"/>
	</node>

</launch>